### ISSUE TITLE
various issues with LMS add-on: frequent 404 then add-on won't work

### ISSUE THREAD

**Issue by marcgarciamarti** [pssc/ha-addon-lms#81]
Hello everyone,

I just installed this add-on today and I must have done something wrong because it has started to act up on me to the point that now I simply can't get it to start, regardless of what I do.

When I first installed it, I tried to change the visualization plugin and that must have corrupted some configuration, as any attempt I made from there on, led me to see multiple 404 error messages. I went ahead and uninstalled the add-on and reinstalled it back. The 404 never went away and the visualization was very very basic, which suggests that I had already ruined some of the base configuration. Uninstalled it and reinstalled it again, but it now won't start:

<img width="1074" alt="Screenshot 2024-05-18 at 12 01 02" src="https://github.com/pssc/ha-addon-lms/assets/12695572/ab2b9913-8d7b-4f9e-b412-6db912ba7fe0">

The rest of my HA configuration works just fine.

Is there any way I can wipe the configuration of the add-on, and start from scratch?

thanks!!

**Comment by marcgarciamarti** [comment]
Things are getting more and more interesting. I now have managed to start the add-on but it won't display LMS anymore. Instead, I see one other add-on!

Note that I have selected "LMS" on the left blade:
<img width="1440" alt="Screenshot 2024-05-18 at 12 06 53" src="https://github.com/pssc/ha-addon-lms/assets/12695572/af5767d6-e67f-45e4-9ffe-1d1ca9f581f2">
the snapshot above was taken using Safari and if I go to Chrome
<img width="1435" alt="Screenshot 2024-05-18 at 12 09 08" src="https://github.com/pssc/ha-addon-lms/assets/12695572/0d88337f-9a3b-4abb-8c6b-d0f802dff322">

**Comment by marcgarciamarti** [comment]
```
TERM=xterm-256color
PIP_DISABLE_PIP_VERSION_CHECK=1
SUPERVISOR_TOKEN=083a08d600f5f3616d72ea304018f1b7d9800117ee50cd263c0813125523782794feb590c9d4d8284a5e93cf4a3d3e64d854162fd2acca51
SHLVL=1
LMS_HACFGDIR=/config/lms/addon
LMS_PLUGS=/config/lms/plugins
PYTHONDONTWRITEBYTECODE=1
LMS_CFG=/config/lms
LMS_OPTIONS= 
cont-init: info: /etc/cont-init.d/99-print-env.sh exited 0
PATH=/command:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
LMS_GROUP=nogroup
S6_CMD_WAIT_FOR_SERVICES=1
PIP_NO_CACHE_DIR=1
DEBIAN_FRONTEND=noninteractive
HAA_NAME=lms
_=/usr/bin/env
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun ha-intergration (no readiness notification)
services-up: info: copying legacy longrun lms (no readiness notification)
services-up: info: copying legacy longrun nginx (no readiness notification)
Start Logitech Media Server...
[14:47:30] INFO: Waiting for lms before starting NGinx...
[14:47:30] INFO: Waiting for Logitech Media Server
s6-rc: info: service legacy-services successfully started
[24-05-18 14:47:31.2105] main::init (386) Starting Logitech Media Server (v8.5.1, 1713072758, Mon 15 Apr 2024 11:37:12 AM CEST) perl 5.036000 - x86_64-linux-gnu-thread-multi
[24-05-18 14:47:31.5547] Slim::Plugin::FullTextSearch::Plugin::_initPopularTerms (592) Fulltext index missing or outdated - re-building
[24-05-18 14:47:31.5548] Slim::Plugin::FullTextSearch::Plugin::_rebuildIndex (480) Starting fulltext index build
[24-05-18 14:47:31.5549] Slim::Plugin::FullTextSearch::Plugin::_rebuildIndex (488) Initialize fulltext table
[24-05-18 14:47:31.5556] Slim::Plugin::FullTextSearch::Plugin::_rebuildIndex (501) Create fulltext index for tracks
[24-05-18 14:47:31.5559] Slim::Plugin::FullTextSearch::Plugin::_rebuildIndex (511) Create fulltext index for albums
[24-05-18 14:47:31.5560] Slim::Plugin::FullTextSearch::Plugin::_rebuildIndex (520) Create fulltext index for contributors
[24-05-18 14:47:31.5564] Slim::Plugin::FullTextSearch::Plugin::_rebuildIndex (530) Create fulltext index for playlists
[24-05-18 14:47:31.5575] Slim::Plugin::FullTextSearch::Plugin::_rebuildIndex (545) Optimize fulltext index
[24-05-18 14:47:31.5578] Slim::Plugin::FullTextSearch::Plugin::_rebuildIndex (560) Fulltext index build done!
[24-05-18 14:47:31.5701] main::checkDataSource (1109) Warning: Schema updated or no media found in the database, initiating scan.
[24-05-18 14:47:31.8351] main::main (205) Starting Logitech Media Server scanner (v8.5.1, 1713072758, Mon 15 Apr 2024 11:37:12 AM CEST) perl 5.036000
[24-05-18 14:47:31.9422] Slim::Music::Import::runImporter (579) Starting Slim::Media::MediaFolderScan scan
[24-05-18 14:47:31.9424] Slim::Utils::Scanner::Local::rescan (179) Discovering audio files in /media
[24-05-18 14:47:31.9775] Slim::Utils::Scanner::Local::__ANON__ (191) Start processing found tracks
[24-05-18 14:47:31.9776] Slim::Utils::Scanner::Local::__ANON__ (199) Connect do DB
[24-05-18 14:47:31.9776] Slim::Utils::Scanner::Local::__ANON__ (202) Get latest ID
[24-05-18 14:47:31.9779] Slim::Utils::Scanner::Local::__ANON__ (227) Build temporary table for deleted tracks
[24-05-18 14:47:31.9783] Slim::Utils::Scanner::Local::__ANON__ (248) Build temporary table for new tracks
[24-05-18 14:47:31.9786] Slim::Utils::Scanner::Local::__ANON__ (268) Build temporary table for changed tracks
[24-05-18 14:47:31.9788] Slim::Utils::Scanner::Local::__ANON__ (291) Get deleted tracks count
[24-05-18 14:47:31.9789] Slim::Utils::Scanner::Local::__ANON__ (298) Get new tracks count
[24-05-18 14:47:31.9791] Slim::Utils::Scanner::Local::__ANON__ (303) Get changed tracks count
[24-05-18 14:47:31.9792] Slim::Utils::Scanner::Local::deleteTracks (420) Removing deleted audio files (0)
[24-05-18 14:47:31.9793] Slim::Utils::Scanner::Local::updateTracks (587) Rescanning changed audio files (0)
[24-05-18 14:47:31.9794] Slim::Utils::Scanner::Local::addTracks (503) Scanning new audio files (0)
[24-05-18 14:47:31.9796] Slim::Music::Import::endImporter (712) Completed Slim::Media::MediaFolderScan Scan in 0.037 seconds.
[24-05-18 14:47:31.9800] Slim::Music::Import::runImporter (579) Starting Slim::Music::PlaylistFolderScan scan
[24-05-18 14:47:31.9802] Slim::Utils::Scanner::Local::rescan (179) Discovering audio files in /config/lms/content/playlists
[24-05-18 14:47:31.9841] Slim::Utils::Scanner::Local::__ANON__ (191) Start processing found tracks
[24-05-18 14:47:31.9843] Slim::Utils::Scanner::Local::__ANON__ (199) Connect do DB
[24-05-18 14:47:31.9844] Slim::Utils::Scanner::Local::__ANON__ (202) Get latest ID
[24-05-18 14:47:31.9847] Slim::Utils::Scanner::Local::__ANON__ (227) Build temporary table for deleted tracks
[24-05-18 14:47:31.9851] Slim::Utils::Scanner::Local::__ANON__ (248) Build temporary table for new tracks
[24-05-18 14:47:31.9854] Slim::Utils::Scanner::Local::__ANON__ (268) Build temporary table for changed tracks
[24-05-18 14:47:31.9857] Slim::Utils::Scanner::Local::__ANON__ (291) Get deleted tracks count
[24-05-18 14:47:31.9859] Slim::Utils::Scanner::Local::__ANON__ (298) Get new tracks count
[24-05-18 14:47:31.9861] Slim::Utils::Scanner::Local::__ANON__ (303) Get changed tracks count
[24-05-18 14:47:31.9862] Slim::Utils::Scanner::Local::deleteTracks (420) Removing deleted audio files (0)
[24-05-18 14:47:31.9863] Slim::Utils::Scanner::Local::updateTracks (587) Rescanning changed audio files (0)
[24-05-18 14:47:31.9865] Slim::Utils::Scanner::Local::addTracks (503) Scanning new audio files (0)
[24-05-18 14:47:31.9867] Slim::Music::Import::endImporter (712) Completed Slim::Music::PlaylistFolderScan Scan in 0.007 seconds.
[24-05-18 14:47:31.9870] Slim::Music::Import::runImporter (579) Starting Slim::Plugin::ExtendedBrowseModes::Libraries scan
[24-05-18 14:47:31.9871] Slim::Music::Import::endImporter (712) Completed Slim::Plugin::ExtendedBrowseModes::Libraries Scan in 0.000 seconds.
[24-05-18 14:47:31.9873] Slim::Music::Import::runImporter (579) Starting Slim::Plugin::FullTextSearch::Plugin scan
[24-05-18 14:47:31.9876] Slim::Plugin::FullTextSearch::Plugin::_rebuildIndex (480) Starting fulltext index build
[24-05-18 14:47:31.9877] Slim::Plugin::FullTextSearch::Plugin::_rebuildIndex (488) Initialize fulltext table
[24-05-18 14:47:31.9881] Slim::Plugin::FullTextSearch::Plugin::_rebuildIndex (501) Create fulltext index for tracks
[24-05-18 14:47:31.9908] Slim::Plugin::FullTextSearch::Plugin::_rebuildIndex (511) Create fulltext index for albums
[24-05-18 14:47:31.9912] Slim::Plugin::FullTextSearch::Plugin::_rebuildIndex (520) Create fulltext index for contributors
[24-05-18 14:47:31.9916] Slim::Plugin::FullTextSearch::Plugin::_rebuildIndex (530) Create fulltext index for playlists
[24-05-18 14:47:31.9929] Slim::Plugin::FullTextSearch::Plugin::_rebuildIndex (545) Optimize fulltext index
[24-05-18 14:47:31.9955] Slim::Plugin::FullTextSearch::Plugin::_rebuildIndex (560) Fulltext index build done!
[24-05-18 14:47:31.9958] Slim::Music::Import::endImporter (712) Completed Slim::Plugin::FullTextSearch::Plugin Scan in 0.008 seconds.
[24-05-18 14:47:31.9964] Slim::Music::Artwork::precacheAllArtwork (657) Starting precacheArtwork for 0 albums
[24-05-18 14:47:31.9966] Slim::Music::Import::endImporter (712) Completed precacheArtwork Scan in 0.000 seconds.
[24-05-18 14:47:31.9969] Slim::Music::Import::runScanPostProcessing (481) Starting Database optimization.
[24-05-18 14:47:32.0133] Slim::Music::Import::endImporter (712) Completed dbOptimize Scan in 0.016 seconds.
[14:48:30] INFO: Start sensor loop for pugin
[14:48:30] INFO: Event lms.service
{"message":"Event lms.service fired."}
[14:48:30] INFO: Starting the Home Assistant STDIN service for lms...
{"entity_id":"binary_sensor.entity_state_cli_test","state":"on","attributes":{},"last_changed":"2024-05-18T10:01:47.054012+00:00","last_reported":"2024-05-18T10:01:47.054012+00:00","last_updated":"2024-05-18T10:01:47.054012+00:00","context":{"id":"01HY5JCZSEGW9F7VV717XXNRY1","parent_id":null,"user_id":"faee409674c64715b540a1d80d1f1668"}}
malformed JSON string, neither array, object, number, string or atom, at character offset 0 (before "Can't connect to 127...") at /usr/share/perl5/JSON/RPC/Legacy/Client.pm line 180.
{"entity_id":"sensor.lms_players","state":"count","attributes":{},"last_changed":"2024-05-18T10:01:47.164899+00:00","last_reported":"2024-05-18T10:01:47.164899+00:00","last_updated":"2024-05-18T10:01:47.164899+00:00","context":{"id":"01HY5JCZWWVKDK9NDENNRC8489","parent_id":null,"user_id":"faee409674c64715b540a1d80d1f1668"}}
malformed JSON string, neither array, object, number, string or atom, at character offset 0 (before "Can't connect to 127...") at /usr/share/perl5/JSON/RPC/Legacy/Client.pm line 180.
{"entity_id":"sensor.lms_albums","state":"count","attributes":{},"last_changed":"2024-05-18T10:01:47.258455+00:00","last_reported":"2024-05-18T10:01:47.258455+00:00","last_updated":"2024-05-18T10:01:47.258455+00:00","context":{"id":"01HY5JCZZTG35B23QK707V2W4P","parent_id":null,"user_id":"faee409674c64715b540a1d80d1f1668"}}
malformed JSON string, neither array, object, number, string or atom, at character offset 0 (before "Can't connect to 127...") at /usr/share/perl5/JSON/RPC/Legacy/Client.pm line 180.
{"entity_id":"sensor.lms_songs","state":"count","attributes":{},"last_changed":"2024-05-18T10:01:47.350716+00:00","last_reported":"2024-05-18T10:01:47.350716+00:00","last_updated":"2024-05-18T10:01:47.350716+00:00","context":{"id":"01HY5JD02P5GGT1QHB6E3PKKXH","parent_id":null,"user_id":"faee409674c64715b540a1d80d1f1668"}}
malformed JSON string, neither array, object, number, string or atom, at character offset 0 (before "Can't connect to 127...") at /usr/share/perl5/JSON/RPC/Legacy/Client.pm line 180.
{"entity_id":"sensor.lms_genres","state":"count","attributes":{},"last_changed":"2024-05-18T10:01:47.443561+00:00","last_reported":"2024-05-18T10:01:47.443561+00:00","last_updated":"2024-05-18T10:01:47.443561+00:00","context":{"id":"01HY5JD05K4D9XPT4FCQK8QHJJ","parent_id":null,"user_id":"faee409674c64715b540a1d80d1f1668"}}
malformed JSON string, neither array, object, number, string or atom, at character offset 0 (before "Can't connect to 127...") at /usr/share/perl5/JSON/RPC/Legacy/Client.pm line 180.
{"entity_id":"sensor.lms_artists","state":"count","attributes":{},"last_changed":"2024-05-18T10:01:47.536817+00:00","last_reported":"2024-05-18T10:01:47.536817+00:00","last_updated":"2024-05-18T10:01:47.536817+00:00","context":{"id":"01HY5JD08GH6Y5XX14T6KWHVH3","parent_id":null,"user_id":"faee409674c64715b540a1d80d1f1668"}}
malformed JSON string, neither array, object, number, string or atom, at character offset 0 (before "Can't connect to 127...") at /usr/share/perl5/JSON/RPC/Legacy/Client.pm line 180.
{"entity_id":"sensor.lms_duration","state":"seconds","attributes":{},"last_changed":"2024-05-18T10:01:47.630056+00:00","last_reported":"2024-05-18T10:01:47.630056+00:00","last_updated":"2024-05-18T10:01:47.630056+00:00","context":{"id":"01HY5JD0BE158YAXXKZPBQ2M20","parent_id":null,"user_id":"faee409674c64715b540a1d80d1f1668"}}
```

**Comment by pssc** [comment]
Hi looks like the addon is running fine  my guess is you have something else running on port 9000 its probably com up and configured itself on a different port... that will now be burt into the config.

**Comment by pssc** [comment]
@marcgarciamarti fixed?

**Comment by marcgarciamarti** [comment]
> @marcgarciamarti fixed?

Yep, it's running fine. I had defined a custom port on the HA frontend, but that setting seems to not be taken into consideration. 

As you very well pointed out, portainer was running on port 9000 and given that the custom LMS port was not being used, the two add-on were clashing. I discovered in the LMS config file that port 9001 was being used by LMS, so I did what you suggested:
- moved portainer to a custom port
- edited the LMS config file and set port to 9000 there

thanks for asking and thanks for the guidance!!
